docs: Add note about using project deployments with catalog apps

cbuto opened the pull request:
<!--
NOTE: You can preview the docs build using the following URL, adding the PR # where specified:
http://docs-d2iq-com-pr-<add_pr_##_here>.s3-website-us-west-2.amazonaws.com/
-->

## Jira Ticket

<!-- Before creating this pull request, make sure you have a separate ticket for the docs team to track their work. If you need assistance, ask in the #documentation Slack channel. -->

https://jira.d2iq.com/browse/D2IQ-83463

## Description of changes being made

Update catalog apps docs to recommend users to use Project Deployments to manage their custom resources across all clusters in a project. I tested this with the follow repo https://github.com/cbuto/gitops-example

This will also need to be carried over to 2.2.

I'm really open to suggestions on a better placement for this and if I should add a more detailed example of using Project Deployment with catalog apps. 

## Checklist

- [ ] Test all commands and procedures, if applicable.
- [ ] Create new PRs against `develop`, or `main` for hotfixes to production.
- [ ] Update all links if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> Example: `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/blob/main/CONTRIBUTING.md) for more information.

mwrightd2 reviewed (dismissed):
Yeah we'll need to get this into develop as well. Looks good though from a language/org perspective within 2.1, simple enough additions.

cwyl02 commented on pages/dkp/kommander/2.1/projects/applications/catalog-applications/dkp-applications/custom-resources-workspace-catalog/kafka/index.md:
```diff
@@ -14,7 +14,7 @@ After deploying the Kafka operator, create Kafka Clusters by applying a `KafkaCl
 ## Example deployment
 
-<p class="message--note"><strong>NOTE: </strong>The Custom Resources must be applied to each attached cluster in a project.</p>
+<p class="message--note"><strong>NOTE: </strong>If you need to manage these custom resources across all clusters in a project, it is recommended you use [Project Deployments](../../../../../project-deployments) which enables you to leverage GitOps to deploy the resources. Otherwise, you will need to create the custom resources manually in each cluster.</p>
```
hmm, for some reasons the link is not working when I try running local dev. it displays as `[Project Deployments](../../../../../project-deployments)`. maybe within the html tags it just ignore markdown syntax? similar for the other 2 apps

cbuto replied:
good catch! let me fix that

takirala commented:
Looking at the acceptance criteria:

> Ability to create GitopsRepository to manage project CRs like SparkApplication.
> Should install/uninstall spark if/when a cluster is added/removed from a project.

Have we tested that these work? (i think they should, but just making sure if there are any nuances.

takirala commented on pages/dkp/kommander/2.1/projects/project-deployments/continuous-delivery/index.md:
```diff
@@ -11,6 +11,12 @@ excerpt: Continuous Deployment
 
 After installing Kommander and [configuring your project and its clusters](../../), navigate to the **Continuous Deployment (CD)** tab under your Project. Here you create a GitOps source which is a source code management (SCM) repository hosting the application definition. D2iQ recommends that you create a secret first then create a GitOps source accessed by the secret.
 
+## Create a Git repository
```
this is ambiguous. there are multiple meanings for this - `github repository`, `any general git repository`, `gitops repository` and then there are CRDs named `GitRepository`, `GitopsRepository` etc., It would be nice to clarify this if possible.

cbuto replied:
good point, this should really be any general git repository but not sure how to specify that 🤔. I'll think on it

JohnB64-JBV replied:
The next paragraph, and the next heading after this one, seem to imply that the customer is creating a GitOps repo....

JohnB64-JBV replied:
You could create a heading that says something on the order of “Create a general Git repository for application files," if that's accurate. Then your first para sentence 1 should probably read: "To create a GitOps source, you first need to create a general Git repository to contain your application definitions and Flux resources."

cbuto replied:
I like that approach, more precise. what do you think about e0711f60db6b4ad29016e74b9d26d334459de90a?

cbuto commented:
> Looking at the acceptance criteria:
> 
> > Ability to create GitopsRepository to manage project CRs like SparkApplication.
> > Should install/uninstall spark if/when a cluster is added/removed from a project.
> 
> Have we tested that these work? (i think they should, but just making sure if there are any nuances.

yep! I put a comment on the ticket but I tested on the daily cluster using this simple repo https://github.com/cbuto/gitops-example. I created a project in the management cluster workspace and added that repo as a GitOps source, the zk cluster was created on the cluster and also deleted if I removed the cluster from the project.

cmastr commented:
@cbuto if this is a 2.2 change then it ONLY needs to go into 2.2 on the develop branch, otherwise this will go to production docs before the release is GA'd. Can you move this?

cbuto commented:
> @cbuto if this is a 2.2 change then it ONLY needs to go into 2.2 on the develop branch, otherwise this will go to production docs before the release is GA'd. Can you move this?

This works with 2.1.1 as well since this was an existing feature and no changes were need to Kommander to support this, I can move it if we don't want it in the 2.1 docs.

cbuto commented:
we discussed this a bit in standup and had some questions that we wanted to bring up here (hoping to get some more input)

1. Should we create a more detailed walkthrough on how to use Project Deployments specifically for catalog applications custom resources? This would involve something like an example repo and steps to use it. 

2. Should we entirely replace the manual steps for all catalog application custom resources (i.e. these [steps](https://docs.d2iq.com/dkp/kommander/2.1/projects/applications/catalog-applications/dkp-applications/custom-resources-workspace-catalog/kafka/#example-deployment)) with a generic Project Deployment example? We would likely have to still include references for example custom resources but the team thinks we should point customers away from manual steps entirely and only use Project Deployments.

mwrightd2 commented:
> we discussed this a bit in standup and had some questions that we wanted to bring up here (hoping to get some more input)
> 
> 1. Should we create a more detailed walkthrough on how to use Project Deployments specifically for catalog applications custom resources? This would involve something like an example repo and steps to use it.
> 2. Should we entirely replace the manual steps for all catalog application custom resources (i.e. these [steps](https://docs.d2iq.com/dkp/kommander/2.1/projects/applications/catalog-applications/dkp-applications/custom-resources-workspace-catalog/kafka/#example-deployment)) with a generic Project Deployment example? We would likely have to still include references for example custom resources but the team thinks we should point customers away from manual steps entirely and only use Project Deployments.

I'm all for creating a one pager that has specific references to each of the custom resource types but to me that's an entirely separate PR/JIRA ticket with a longer window. At this time I think adding notes to the custom resource pages as they stand is fine as a backmerge for 2.1.1 and we can spin something up for 2.2 GA. 

Since doc work for that is inflight we'd need to sit down soon and discuss the granular stuff concerning this new Project Deployment walkthrough. My 2c, @cmastr?
